### DISCUSSION TITLE
Name suggestions for Open ended

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/flow#83] [Ideas]
Seems like we might need a new name for open ended.

Suggestion by @PaulRBerg:

- Continuous
- Flow
- FlexFlow
- Everflow
- Progressive
- TopUp

Suggestions by @smol-ninja:
- Unlock (opposite of locked which is lockup)
- OpenFlow
- Infinio
- Pulse
- Eien (its a Japanese word for eternity)

Once again, request for new names @sablier-labs/everybody.

**Comment by PaulRBerg** [comment]
My top favorites now, ordered by preference:

- Everflow
- Infinio
- Flow
- OpenFlow
- Continuous
- FlexFlow

I don't like the others:

- _Unlock_ gives vesting vibes; consider that people use this word to refer to portions that get unlocked during a vesting period
- _Eien_ is interesting, but it would 100% be misspelled; it doesn't pass the party test
- _Pulse_ is boring and used by a dozen other projects
- _TopUp_ sounds like an escrow service or a credit card service
- _Progressive_ is vague; also, it may be confused for a non-linear distribution

**Reply by smol-ninja** [reply]
Agree with your explanations about disliking the other names (Eien can also easily be mispronounced. I have been mispronouncing it lol [yt video](https://www.youtube.com/watch?v=Rc0VACs7dow)). From the ones you liked, my order preference would be:

- Infinio
- Flow
- Flex (I like this more then Flexflow)
- Openflow
- Continuous or Continuum
- Everflow
- Flexflow

**Comment by maxdesalle** [comment]
My personal preferences goes to Flow, as it's short and effective. Other names are either too long, or composed of two words ("Open-Flow"), which isn't ideal.

**Reply by PaulRBerg** [reply]
Yeah, Flow is pretty good, definitely better than `Continuous` (which is vague) and `OpenFlow` and `FlexFlow` (which are indeed two-word structures).

But I wouldn't call `Infinio` Or `Everflow` too long.

_Everflow_ isn't composed of two words. _Everflowing_ is actually a [word](https://www.yourdictionary.com/everflowing) - it means "which flows unceasingly".

**Reply by PaulRBerg** [reply]
IMO Everflow is better than Flow because it'd be a more special name. There are dozens of crypto projects with the word 'flow' in them, e.g. [this $FLOW token](https://www.coingecko.com/en/coins/flow).

**Reply by maxdesalle** [reply]
Still prefer Flow given it will be very very often prefixed by "Sablier", and "Sablier EverFlow" is too long.

**Reply by PaulRBerg** [reply]
Good point

**Comment by PaulRBerg** [comment]
Tagging @sablier-labs/everybody for their feedback.

The leading candidates thus far:

- Flow
- Everflow
- Infinio
- Flex

**Reply by razgraf** [reply]
Given our existing product is called `Lockup` I think a simple name fits best - they'll be part of the same product lineup, so sounding good near "Sablier Lockup and ___" is important to consider - so good final choices you've selected.

I like Flow / Flex.

---

There's also "Sablier Unlimited" (got it from ChatGPT as a synonym to "Open Ended") which sounds really really good hehe (works for the lineup too: "Lockup and Unlimited") but I fear it's a bit to grandiose.

**Reply by PaulRBerg** [reply]
Good point about `Lockup` matching.

But what's wrong with `Sablier Lockup and Everflow`? :D

> "Sablier Unlimited" ... it's a bit to grandiose.

It is.

**Reply by razgraf** [reply]
Nothing I guess, I just like Flow because it's shorter.

**Comment by andreivladbrg** [comment]
My favorites from the ones already mentioned are:
 - `Flow`
- `Infinio` - this one may be hard to spell

One other suggestion (also from GPT): `SablierElastic` - captures the idea of elasticity in funding and time constraints

**Reply by smol-ninja** [reply]
I like Sablier Elastic as well. (Sablier Lockup and Sablier Elastic).

**Reply by PaulRBerg** [reply]
In a development context, Elastic is typically used to refer to Elastic computing, e.g. AWS's EC2.

**Comment by gavriliumircea** [comment]
My favorites for now are:

- Flow
- Elastic

And I think there are a good match with Lockup as well.

**Reply by PaulRBerg** [reply]
In a development context, Elastic is typically used to refer to Elastic computing, e.g. AWS's EC2.

**Comment by IaroslavMazur** [comment]
Great job, @andreivladbrg, on discovering "Elastic"! I like how it sounds and what meaning it conveys (the product's user-friendliness and adaptability for the users' needs, its flexibility, etc.)

Speaking of which... Wdyat about "Flex"? 👀

"Sablier Flex" is concise, yet meaningful, while "Sablier Lockup & Flex" seem to get along well, too, and kinda outline their differences from the get-go.

**Reply by PaulRBerg** [reply]
In a development context, Elastic is typically used to refer to Elastic computing, e.g. AWS's EC2.

**Comment by andreivladbrg** [comment]
What do you guys think about `Elastic ` @razgraf @PaulRBerg ?

Should we create a poll?

**Reply by PaulRBerg** [reply]
I don't like it all, it doesn't hint at this contract being about continuous payments.

In a development context, Elastic is typically used to refer to Elastic computing, e.g. AWS's EC2.

Good idea to create a poll, I'll do it now.

**Comment by PaulRBerg** [comment]
Created a poll here, @sablier-labs/everybody can you please vote:

https://github.com/sablier-labs/v2-open-ended/discussions/94

Closing this discussion in favor of the poll.
